storage/watchable_store_bench_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"math/rand"
+	"os"
+	"testing"
+)
+
+func BenchmarkWatchableStoreUnsyncedCancel(b *testing.B) {
+	watcherSize := b.N
+	s := &watchableStore{
+		KV:       newStore(tmpPath),
+		unsynced: make(map[*watcher]struct{}),
+		synced:   make(map[string][]*watcher),
+	}
+
+	defer func() {
+		s.KV.Close()
+		os.Remove(tmpPath)
+	}()
+
+	b.ReportAllocs()
+
+	testKey := []byte("foo")
+	testValue := []byte("bar")
+	s.Put(testKey, testValue)
+
+	cancels := []CancelFunc{}
+	// create 2*N watchers
+	for i := 0; i < 2*watcherSize; i++ {
+		_, cancel := s.Watcher(testKey, true, -1)
+		cancels = append(cancels, cancel)
+	}
+
+	// random-cancel N watchers to make it not biased towards
+	// data structures with an order, such as slice.
+	idxSlice := rand.Perm(2 * watcherSize)
```

Comment:
idxslice is a terrible name in my opinion. It simple means nothing. If you do not care about its naming, just use two chars. If you do care about naming, include the fact that it is a rand perm at least.